src/jv_unicode.c

```diff
@@ -87,7 +87,8 @@ int jvp_utf8_decode_length(char startchar) {
 	if ((startchar & 0x80) == 0) return 1;         // 0___ ____
 	else if ((startchar & 0xE0) == 0xC0) return 2; // 110_ ____
 	else if ((startchar & 0xF0) == 0xE0) return 3; // 1110 ____
-	else return 4;                                 // 1111 ____
+	else if ((startchar & 0xF0) == 0xF0) return 4; // 1111 ____
```

Comment:
Note that this change isn't needed in order to fix #1804.  This function correctly decoded the length of the UTF-8-encoded `U+FFFD` as `3`.

Reply:
Actually, this commit, I think is not needed at all.  At 88 we know at least the high bit is set, and if it's actually two are set, we return 2.  At 89 we know at least the two high bits are set, and if it's actually three, then we return three.  At 90 it can only be that the four high bits are set, so the new condition at 90 is always true.

Reply:
Doesn't it skip the 10__ ____ case if a broken continuation byte sneaks in? But assuming that the UTF-8 is well-formed, yes, this change is unnecessary.

Reply:
I take your point.  Perhaps `f_match()` should use `jvp_utf8_next()` and re-encode it.  But as long as only `f_match()` calls this, and `f_match()` checks that there are as many bytes left in the put as this function returns, then it's safe.  If standard C had local functions...